text/0814-stylelint.md

```diff
@@ -0,0 +1,180 @@
+over 170 built-in rules, sane default configs, and robust plugin support.
+
+## Detailed design
+
+The general idea is that we will update the `app` and `addon` [blueprints](https://github.com/ember-cli/ember-cli/tree/master/blueprints) 
+to add a few Stylelint-related packages to the `package.json`, add config files to enable this tooling 
+by default, and add commands to the `scripts` section of `package.json` to expose easy access to this 
+tool consistent with our existing linters.
+
+### Packages
+
+The following packages will be added to the `devDependencies` section of both blueprints:
+
+* [stylelint](https://github.com/stylelint/stylelint) - The base linter
+* [stylelint-config-standard](https://github.com/stylelint/stylelint-config-standard) - The standard rule config supported by the stylelint community
+* [stylelint-prettier](https://github.com/prettier/stylelint-prettier) - Plugin to support prettier auto-formatting as stylelint rules
+* [stylelint-config-prettier](https://github.com/prettier/stylelint-config-prettier) - Plugin to disable stylelint rules that conflict with prettier's automatic behavior
+
+Prettier itself is already included in the Ember blueprints by default.
+
+### Configuration Changes
+
+A `.stylelintrc.js` file will be added with the following contents:
+
+```js title=".stylelintrc.js"
+'use strict';
+
+module.exports = {
+  extends: ['stylelint-config-standard', 'stylelint-prettier/recommended'],
+};
+```
+
+There are many rules, configs, and plugins available for stylelint and prettier, but we are 
+intentionally avoiding being prescriptive on anything that isn't nearly universally agreed upon 
+by the stylelint, prettier, and Ember communities.
+
+#### Lint Ignores
+
+A `.stylelintignore` file will be added with the following contents:
+
+```title=".stylelintignore"
+# unconventional js
+/blueprints/*/files/
+/vendor/
+
+# compiled output
+/dist/
+/tmp/
+
+# dependencies
+/bower_components/
+/node_modules/
+
+# misc
+/coverage/
+!.*
+.*/
+.eslintcache
+.lint-todo/
+
+# ember-try
+/.node_modules.ember-try/
+/bower.json.ember-try
+/npm-shrinkwrap.json.ember-try
+/package.json.ember-try
+/package-lock.json.ember-try
+/yarn.lock.ember-try
+```
+
+These contents are based on the existing `.eslintignore` and `.prettierignore` files for consistency.
+
+### Blueprint Changes
+
+In general, it is recommended that all blueprints provided by addons should satisfy the default 
+linting configuration of a new Ember application. As such, the blueprints provided by ember-cli 
+will be updated as needed to ensure that they satisfy these new linting requirements.
+
+#### `package.json` scripts
+
+The `app` and `addon` blueprints will be updated to add style-related linting scripts:
+
+```json title="package.json"
+{
+  "scripts": {
+    "lint:css": "stylelint \"**/*.{css}\"",
+    "lint:css:fix": "stylelint \"**/*.{css}\" --fix",
```

Comment:
To avoid repeating this stylelint command/glob in two scripts, we can use this:

```suggestion
    "lint:css:fix": "npm-run-all \"lint:css --fix\"",
```

npm-run-all which is already used in other scripts allows us to avoid directly referencing npm or yarn (as we don't know which one the user is using).